modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,233 @@
+:FeatureName: The Shared Resources CSI Driver
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+:FeatureName: TBD
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+The Shared Resources CSI Driver and TBD features also belong to the `TechPreviewNoUpgrade` feature set, which is a subset of the current Technology Preview features. You can enable the `TechPreviewNoUpgrade` feature set on test clusters, where you can fully test them while leaving the features disabled on production clusters. Enabling this feature set cannot be undone and prevents updates. This feature set is not recommended on production clusters. See "Enabling Technology Preview features using feature gates" in the following "Additional resources" section.
+====
+
+.User roles
+
+The following procedure includes several types of users:
+
+* An _administrator_ with cluster-level permissions to access multiple namespaces, create cluster-scoped objects, and create `Role` and `RoleBinding` objects around those cluster-scoped objects, both for discovery and use.
+* Optionally, _namespace administrators_ can perform some of the tasks in this example, such as creating  `Role` and `RoleBinding` objects for users or service accounts.
+* General users of the cluster, such as  _developers_ or _testers_, are generally restricted to access a single namespace to build images with OpenShift Builds.
++
+So that these users can use RHEL entitlements in their builds, administrators give them access to discover `SharedSecret` and `SharedConfigMaps` custom resource (CR) instances that they can reference in their `BuildConfig` objects. Also, their namespace `builder` service account has been permitted to mount the specific `SharedSecret` CR instance as a Shared Resource CSI Driver volume in OpenShift Build pods.
+
+.Prerequisites
+
+* You have enabled the  `TechPreviewNoUpgrade` feature set by using the feature gates.
+* You have created and registered a `Secret` object at https://console.redhat.com/.
+
+.Procedure
+
+. As a cluster or namespace administrator, create a `Namespace` object to host your `BuildConfig` and `Build` objects.
++
+.Example YAML Namespace object definition
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-csi-app-namespace
+----
+
+. As a cluster administrator, create a `SharedSecret` custom resource (CR) instance that references the `Secret` object where the Insights Operator stores the subscription credentials for the cluster.
++
+.Example YAML `SharedSecret` custom resource definition
+[source,yaml]
+----
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: my-share-bc
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+----
+
+. As a cluster or namespace administrator, create a `Role` object so that the `builder` service account in our namespace can access the `SharedSecret` CR instance.
++
+.Example YAML `Role` object definition
```

Comment:
can you add the tidbit we in the shared resource section where oc/kubectl won't let us do `oc create role`